pubsub/google/cloud/pubsub_v1/subscriber/policy/thread.py

```diff
@@ -159,7 +174,7 @@ def close(self):
 
         # The subscription is closing cleanly; resolve the future if it is not
         # resolved already.
-        if self._future is not None and not self._future.done():
+        if not self._future.done():
             self._future.set_result(None)
         self._future = None
```

Comment:
@lukesneeringer WDYT of returning the future here?

```python
future = self._future
self._future = None
return future
```

Reply:
That is fine with me.